flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java

```diff
@@ -152,6 +177,38 @@ public void setFields(boolean[] sourceFieldMask, Class<?>[] fieldTypes) {
 	public Class<?>[] getFieldTypes() {
 		return super.getGenericFieldTypes();
 	}
+
+	public void setFieldsMap(String[] fieldsMap) {
+		Preconditions.checkNotNull(fieldsMap);
+		Preconditions.checkState(typeInformation instanceof PojoTypeInfo);
+
+		PojoTypeInfo<OUT> pojoTypeInfo = (PojoTypeInfo<OUT>) typeInformation;
+
+		String[] fields = pojoTypeInfo.getFieldNames();
+		Class<?>[] fieldTypes = getFieldTypes();
+		this.fieldsMap = Arrays.copyOfRange(fieldsMap, 0, fieldsMap.length);
+
+		boolean[] includeMask = new boolean[fieldsMap.length];
+		Class<?>[] newFieldTypes = new Class<?>[fieldsMap.length];
+
+		for (int i = 0; i < fieldsMap.length; i++) {
+			if (fieldsMap[i] == null) {
```

Comment:
IMO, ``null`` values should not be allowed in the ``fieldsMap``. Can you throw an exception in that case?
The ``fieldsMap`` should be a list of fields that are mapped to columns in the CSV file. As said before, the columns that are read by the format are defined by the ``includeMask``.